be/test/vec/function/function_round_test.cpp

```diff
@@ -0,0 +1,1133 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include <climits>
+#include <cmath>
+#include <cstddef>
+#include <map>
+#include <memory>
+#include <tuple>
+#include <utility>
+#include <vector>
+
+#include "function_test_util.h"
+#include "vec/columns/column.h"
+#include "vec/columns/column_const.h"
+#include "vec/columns/column_decimal.h"
+#include "vec/columns/columns_number.h"
+#include "vec/common/assert_cast.h"
+#include "vec/core/column_numbers.h"
+#include "vec/core/types.h"
+#include "vec/data_types/data_type_decimal.h"
+#include "vec/data_types/data_type_number.h"
+#include "vec/functions/round.h"
+
+/**
+ This BE UT focus on enhancement of round based function, which enables them 
+ to use column as scale argument. We test truncate/floor/ceil/round/round_bankers 
+ together by moving test cases of function_truncate_test.cpp here.
+*/
+
+namespace doris::vectorized {
+// {precision, scale} -> {input, scale_arg, expectation}
+using DecimalTestDataSet =
+        std::map<std::pair<int, int>, std::vector<std::tuple<Int128, int, Int128>>>;
+
+// input, scale_arg, expectation
+using FloatTestDataSet = std::vector<std::tuple<Float64, int, Float64>>;
+
+using DecimalTruncateFunction = FunctionRounding<DecimalRoundTwoImpl<TruncateName>,
+                                                 RoundingMode::Trunc, TieBreakingMode::Auto>;
+using DecimalFloorFunction = FunctionRounding<DecimalRoundTwoImpl<FloorName>, RoundingMode::Floor,
+                                              TieBreakingMode::Auto>;
+using DecimalCeilFunction =
+        FunctionRounding<DecimalRoundTwoImpl<CeilName>, RoundingMode::Ceil, TieBreakingMode::Auto>;
```

Comment:
https://dev.mysql.com/doc/refman/8.0/en/mathematical-functions.html#function_ceil

ceil function in mysql only has one argument.

Reply:
Yes, I noticed, but Doris already has two arguments since long time ago so I keep.